core/src/health.rs

```diff
@@ -0,0 +1,214 @@
+    pub fn new(port: u16, manifest: Arc<Manifest>, postgres_client: Option<Arc<PostgresClient>>) -> Self {
+        Self {
+            port,
+            state: HealthServerState {
+                manifest,
+                postgres_client,
+            },
+        }
+    }
+
+    pub async fn start(self) -> Result<(), Box<dyn std::error::Error + Send + Sync>> {
+        let app = Router::new()
+            .route("/health", get(health_handler))
+            .with_state(self.state);
+
+        let addr = SocketAddr::from(([0, 0, 0, 0], self.port));
+        let listener = TcpListener::bind(addr).await?;
+
+        info!("🩺 Health server started on http://0.0.0.0:{}/health", self.port);
+
+        axum::serve(listener, app).await?;
+        Ok(())
+    }
+}
+
+async fn health_handler(State(state): State<HealthServerState>) -> Result<Json<HealthStatus>, StatusCode> {
+    let mut health_status = HealthStatus {
+        status: "healthy".to_string(),
+        timestamp: chrono::Utc::now().to_rfc3339(),
+        services: HealthServices {
+            database: "unknown".to_string(),
+            indexing: "unknown".to_string(),
+            sync: "unknown".to_string(),
+        },
+        indexing: IndexingStatus {
+            active_tasks: active_indexing_count(),
+            is_running: is_running(),
+        },
+    };
+
+    // Check database connection if PostgreSQL is enabled
+    if state.manifest.storage.postgres_enabled() {
+        match &state.postgres_client {
+            Some(client) => {
+                match client.query_one("SELECT 1", &[]).await {
+                    Ok(_) => {
+                        health_status.services.database = "healthy".to_string();
+                    }
+                    Err(e) => {
+                        health_status.services.database = "unhealthy".to_string();
+                        health_status.status = "unhealthy".to_string();
+                        error!("Database health check failed: {}", e);
+                    }
+                }
+            }
+            None => {
+                health_status.services.database = "not_configured".to_string();
+                health_status.status = "unhealthy".to_string();
+            }
+        }
+    } else {
+        health_status.services.database = "disabled".to_string();
+    }
+
+    // Check indexing status
+    if health_status.indexing.is_running {
+        health_status.services.indexing = "healthy".to_string();
+    } else {
+        health_status.services.indexing = "stopped".to_string();
+        health_status.status = "unhealthy".to_string();
+    }
+
+    // Check sync status (basic check for event tables if using PostgreSQL)
+    if state.manifest.storage.postgres_enabled() {
+        match &state.postgres_client {
+            Some(client) => {
+                match client.query_one(
+                    "SELECT COUNT(*) FROM information_schema.tables WHERE table_schema = 'public' AND table_name LIKE '%_events'",
```

Comment:
count(*) is expensive you should just check if it exists and do the SELECT 1 FROM ... LIMIT 1